packages/cli-kit/src/public/node/ruby.ts

```diff
@@ -136,31 +137,24 @@ async function installThemeCheckCLIDependencies(stdout: Writable) {
  * Install RubyCLI and its dependencies
  * Shows a loading spinner if it's the first time installing dependencies
  * or if we are installing a new version of RubyCLI.
+ *
+ * @param stdout - The Writable stream on which to write the standard output.
  */
-async function installCLIDependencies() {
+async function installCLIDependencies(stdout: Writable) {
   const exists = await file.fileExists(shopifyCLIDirectory())
-  const tasks = [
-    {
-      title: 'Installing theme dependencies',
-      task: async () => {
-        const usingLocalCLI2 = Boolean(process.env.SHOPIFY_CLI_2_0_DIRECTORY)
-        await validateRubyEnv()
-        if (usingLocalCLI2) {
-          await bundleInstallLocalShopifyCLI()
-        } else {
-          await createShopifyCLIWorkingDirectory()
-          await createShopifyCLIGemfile()
-          await bundleInstallShopifyCLI()
-        }
-      },
-    },
-  ]
 
-  if (exists) {
-    await tasks[0]!.task()
+  if (!exists) stdout.write('Installing theme dependencies...')
+  const usingLocalCLI2 = Boolean(process.env.SHOPIFY_CLI_2_0_DIRECTORY)
```

Comment:
I'd use the `isTruthy` [function](https://github.com/Shopify/cli/blob/d351a9ea7950e507a10980fd085d49958000b333/packages/cli-kit/src/private/node/environment/utilities.ts#L4) for this since the environment variable can take values that can't be coerced to a boolean.